crates/forge-runner/src/lib.rs

```diff
@@ -207,9 +98,10 @@ fn maybe_save_versioned_program(
     Ok(maybe_versioned_program_path)
 }
 
-fn choose_test_strategy_and_run(
+#[must_use]
+pub fn run_test_case(
```

Comment:
The name is good but `run_test_case` is also a function we use deep down in the runner (it actually runs casm). Can we rename the other function so the names differ? Idk maybe to `execute_test` or sth, not sure, 

[nit] The names can be tweaked a bit in general (e.g. `run_test` -> `run_single_test`/`run_singular_test`)

Reply:
changed to `run_for_test_case`, consistent with higher ones